src/main/java/ConsentStringParser.java

```diff
@@ -0,0 +1,450 @@
+		this.cmpVersion = bits.getInt(CMP_VERSION_OFFSET, CMP_VERSION_SIZE);
+		this.consentScreenID = bits.getInt(CONSENT_SCREEN_SIZE_OFFSET, CONSENT_SCREEN_SIZE);
+		this.consentLanguage = bits.getSixBitString(CONSENT_LANGUAGE_OFFSET, CONSENT_LANGUAGE_SIZE);
+		this.vendorListVersion = bits.getInt(VENDOR_LIST_VERSION_OFFSET, VENDOR_LIST_VERSION_SIZE);
+		this.maxVendorSize = bits.getInt(MAX_VENDOR_ID_OFFSET, MAX_VENDOR_ID_SIZE);
+		this.vendorEncodingType = bits.getInt(ENCODING_TYPE_OFFSET, ENCODING_TYPE_SIZE);
+		for (int i = PURPOSES_OFFSET, ii = PURPOSES_OFFSET + PURPOSES_SIZE; i < ii; i++) {
+			allowedPurposes.add(bits.getBit(i));
+		}
+		if (vendorEncodingType == VENDOR_ENCODING_RANGE) {
+			this.rangeEntries = new ArrayList<RangeEntry>();
+			this.defaultConsent = bits.getBit(DEFAULT_CONSENT_OFFSET);
+			int numEntries = bits.getInt(NUM_ENTRIES_OFFSET, NUM_ENTRIES_SIZE);
+			int currentOffset = RANGE_ENTRY_OFFSET;
+			for (int i = 0; i < numEntries; i++) {
+				boolean range = bits.getBit(currentOffset);
+				currentOffset++;
+				if (range) {
+					int startVendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					int endVendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					rangeEntries.add(new RangeEntry(startVendorId, endVendorId));
+				} else {
+					int vendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					rangeEntries.add(new RangeEntry(vendorId));
+				}
+			}
+		}
+
+	}
+
+	/**
+	 * @return the string passes in the constructor.
+	 *
+	 */
+	public String getConsentString() {
+		return consentString;
+	}
+
+	/**
+	 * @return the {@link Instant} at which the cookie was created
+	 */
+	public Instant getCookieCreated() {
+		return cookieCreated;
+	}
+
+	/**
+	 *
+	 * @return the {@link Instant} at which the cookie was last updated
+	 */
+	public Instant getCookieLastUpdated() {
+		return cookieLastUpdated;
+	}
+
+	/**
+	 *
+	 * @return the version of the cookie format used in this consent string
+	 */
+	public int getVersion() {
+		return version;
+	}
+
+	/**
+	 *
+	 * @return the id of the consent management partner that created this consent string
+	 */
+	public int getCmpID() {
+		return cmpID;
+	}
+
+	/**
+	 * @return the version of the cmp being used
+	 */
+	public int getCmpVersion() {
+		return cmpVersion;
+	}
+
+	/**
+	 *
+	 * @return the id of the string through which the user gave consent in the CMP UI
+	 */
+	public int getConsentScreenID() {
+		return consentScreenID;
+	}
+
+	/**
+	 * @return The two letter ISO639-1 language code in which the CMP asked for consent
+	 */
+	public String getConsentLanguage() {
+		return consentLanguage;
+	}
+
+	/**
+	 *
+	 * @return a list of purpose id's which are permitted according to this consent string
+	 */
+	public List<Integer> getAlllowedPurposes() {
+		List<Integer> purposes = new ArrayList<Integer>();
```

Comment:
Is there any way to cache this on the setting of the Consent String, so you don't have to evaluate every time you call this method?